### DISCUSSION TITLE
Rewrite all commands (if applicable) to use Viewer as sender

### DISCUSSION THREAD

**Discussion by Jakubk15** [EternalCodeTeam/EternalCore#381] [Support] [answered]

**Comment by vLuckyyy** [comment]
Example?

**Comment by Jakubk15** [comment]
Niektóre komendy używają jako sendera Playera, a bardziej uniwersalnym i future-proof rozwiązaniem byłoby użycie viewera

**Comment by Jakubk15** [comment]
np. komendy od container stuff

**Comment by Jakubk15** [comment]
tutaj masz przykład: https://github.com/EternalCodeTeam/EternalCore/blob/master/core/src/main/java/com/eternalcode/core/feature/essentials/container/GrindstoneCommand.java

**Comment by P1otrulla** [comment, marked as answer]
> tutaj masz przykład: https://github.com/EternalCodeTeam/EternalCore/blob/master/core/src/main/java/com/eternalcode/core/feature/essentials/container/GrindstoneCommand.java

Jak ty niby chcesz otwierać inventory za pomocą Viewer'a, jak robisz takie issue to wypisz listę komend do zmiany od razu będzie łatwiej osobie która będzie ten issue robiła.

**Comment by Jakubk15** [comment]
Jest wiele takich komend, wystarczy poprzeglądać klasy. Kiedyś dyskutowałem z Martinem na ten temat i chyba on mówił że właśnie używając Viewera będzie lepiej, a co do twojego pytania to Viewer nie ma metod Playera?

**Reply by vLuckyyy** [reply]
No dobra, dasz jakieś przykłady które obecnie komendy można podmienić na viewera?

**Reply by Jakubk15** [reply]
Może spróbuję zrobić PR na dniach, w przeciwnym razie podeślę po researchu
